components/gitpod-db/go/cost_center.go

```diff
@@ -131,6 +131,35 @@ func getCostCenter(ctx context.Context, conn *gorm.DB, attributionId Attribution
 	return costCenter, nil
 }
 
+func (c *CostCenterManager) IncrementBillingCycle(ctx context.Context, attributionId AttributionID) (CostCenter, error) {
+	cc, err := getCostCenter(ctx, c.conn, attributionId)
+	if err != nil {
+		return CostCenter{}, err
+	}
+	now := time.Now().UTC()
+	if cc.NextBillingTime.Time().After(now) {
+		return CostCenter{}, status.Errorf(codes.FailedPrecondition, "cannot increment billing cycle before next billing time")
+	}
+	billingCycleStart := NewVarCharTime(now)
+	if cc.NextBillingTime.IsSet() {
+		billingCycleStart = cc.NextBillingTime
+	}
+	// All fields on the new cost center remain the same, except for BillingCycleStart, NextBillingTime, and CreationTime
+	newCostCenter := CostCenter{
+		ID:                cc.ID,
+		SpendingLimit:     cc.SpendingLimit,
+		BillingStrategy:   cc.BillingStrategy,
+		BillingCycleStart: billingCycleStart,
+		NextBillingTime:   NewVarCharTime(billingCycleStart.Time().AddDate(0, 1, 0)),
+		CreationTime:      NewVarCharTime(now),
+	}
+	err = c.conn.Save(&newCostCenter).Error
+	if err != nil {
+		return CostCenter{}, err
```

Comment:
```suggestion
		return CostCenter{}, fmt.Errorf("failed to store cost center ID: %s", err)
```
otherwise you don't actually know what failed